REFERENCE.md

```diff
@@ -0,0 +1,1404 @@
+
+For example:
+
+    1-1024
+
+This would cover ports 1 to 1024.
+
+##### `dst_type`
+
+Valid values: [:UNSPEC, :UNICAST, :LOCAL, :BROADCAST, :ANYCAST, :MULTICAST,
+                :BLACKHOLE, :UNREACHABLE, :PROHIBIT, :THROW, :NAT, :XRESOLVE].map { |address_type|
+                [
+                  address_type,
+                  "! #{address_type}".to_sym,
+                  "#{address_type} --limit-iface-in".to_sym,
+                  "#{address_type} --limit-iface-out".to_sym,
+                  "! #{address_type} --limit-iface-in".to_sym,
+                  "! #{address_type} --limit-iface-out".to_sym,
+                ]
+              }.flatten
+
+The destination address type. For example:
+
+    dst_type => ['LOCAL']
+
+Can be one of:
+
+* UNSPEC - an unspecified address
+* UNICAST - a unicast address
+* LOCAL - a local address
+* BROADCAST - a broadcast address
+* ANYCAST - an anycast packet
+* MULTICAST - a multicast address
+* BLACKHOLE - a blackhole address
+* UNREACHABLE - an unreachable address
+* PROHIBIT - a prohibited address
+* THROW - undocumented
+* NAT - undocumented
+* XRESOLVE - undocumented
+
+In addition, it accepts '--limit-iface-in' and '--limit-iface-out' flags, specified as:
+
+    dst_type => ['LOCAL --limit-iface-in']
+
+It can also be negated using '!':
+
+    dst_type => ['! LOCAL']
+
+Will accept a single element or an array.
+
+##### `src_type`
+
+Valid values: [:UNSPEC, :UNICAST, :LOCAL, :BROADCAST, :ANYCAST, :MULTICAST,
+                :BLACKHOLE, :UNREACHABLE, :PROHIBIT, :THROW, :NAT, :XRESOLVE].map { |address_type|
+                [
+                  address_type,
+                  "! #{address_type}".to_sym,
+                  "#{address_type} --limit-iface-in".to_sym,
+                  "#{address_type} --limit-iface-out".to_sym,
+                  "! #{address_type} --limit-iface-in".to_sym,
+                  "! #{address_type} --limit-iface-out".to_sym,
+                ]
+              }.flatten
+
+The source address type. For example:
+
+    src_type => ['LOCAL']
+
+Can be one of:
+
+* UNSPEC - an unspecified address
+* UNICAST - a unicast address
+* LOCAL - a local address
+* BROADCAST - a broadcast address
+* ANYCAST - an anycast packet
+* MULTICAST - a multicast address
+* BLACKHOLE - a blackhole address
+* UNREACHABLE - an unreachable address
+* PROHIBIT - a prohibited address
+* THROW - undocumented
+* NAT - undocumented
+* XRESOLVE - undocumented
+
+In addition, it accepts '--limit-iface-in' and '--limit-iface-out' flags, specified as:
+
+    src_type => ['LOCAL --limit-iface-in']
+
+It can also be negated using '!':
+
+    src_type => ['! LOCAL']
+
+Will accept a single element or an array.
+
+##### `proto`
+
+Valid values: [:ip, :tcp, :udp, :icmp, :"ipv6-icmp", :esp, :ah, :vrrp, :igmp, :ipencap, :ipv4, :ipv6, :ospf, :gre, :cbt, :sctp, :pim, :all].map { |proto|
+      [proto, "! #{proto}".to_sym]
+    }.flatten
+
+The specific protocol to match for this rule. By default this is
```

Comment:
This should be removed as default is already specified.

Reply:
Removed